cartography/intel/azure/sql.py

```diff
@@ -0,0 +1,855 @@
+import logging
+from typing import Any
+from typing import Dict
+from typing import Generator
+from typing import List
+from typing import Tuple
+
+import neo4j
+from azure.core.exceptions import HttpResponseError
+from azure.mgmt.sql import SqlManagementClient
+from msrestazure.azure_exceptions import CloudError
+
+from .util.credentials import Credentials
+from cartography.util import run_cleanup_job
+from cartography.util import timeit
+
+logger = logging.getLogger(__name__)
+
+
+@timeit
+def get_client(credentials: Credentials, subscription_id: str) -> SqlManagementClient:
+    """
+    Getting the Azure SQL client
+    """
+    client = SqlManagementClient(credentials, subscription_id)
+    return client
+
+
+@timeit
+def get_server_list(credentials: Credentials, subscription_id: str) -> List[Dict]:
+    """
+    Returning the list of Azure SQL servers.
+    """
+    try:
+        client = get_client(credentials, subscription_id)
+        server_list = list(map(lambda x: x.as_dict(), client.servers.list()))
```

Comment:
Any particular reason to use `list(map(...))`, rather than list comprehension? I wouldn't block but list comprehension reads quicker.

Reply:
@ramonpetgrave64  We wanted to keep the code consistent throughout all Azure modules and since we used `list(map(...))` in the compute.py code. What I'll do is I'll open a PR in the future and change all the `list(map(...))` codes to list comprehension across all Azure modules we'll add. :)